includes/configure/replace.yaml

```diff
@@ -1,69 +1,78 @@
 ---
-# replace.yaml
+# this block is reponsible for performing a configure replace on the target
+# device.  Unlinke configure terminal, the configuration file is loaded wholly
+# into the device and therefore is not seen line by line in the output.
 #
-- name: disable prompt
-  cli:
-    command: terminal dont-ask
-
-- name: loading configuration onto device
+# Note: This feature is only supported on the Nexus 9K platforms.
+#
+- name: load configuration using configure replace
   block:
-  - name: nxos dir list
-    cli:
-      command: dir
-    register: nxos_dir_listing
+    - name: disable prompt
+      cli:
+        command: terminal dont-ask
+
+    - name: loading configuration onto device
+      block:
+        - name: nxos dir list
+          cli:
+            command: dir
+          register: nxos_dir_listing
+
+        - name: create temp working dir
+          tempfile:
+            state: directory
+          register: nxos_temp_working_dir
+          check_mode: false
 
-  - name: create temp working dir
-    file:
-      path: "{{ nxos_config_working_dir }}"
-      state: directory
-    run_once: true
-    check_mode: false
+          # always set check mode to false here as this task needs to always run
+          # in order to fully execute the task list.
+        - name: template source config
+          copy:
+            content: "{{ nxos_config_text }}"
+            dest: "{{ nxos_config_working_dir.path }}/{{ nxos_config_temp_config_file }}"
```

Comment:
`nxos_config_working_dir` is missing in defaults/main.yml

Reply:
@trishnaguha take a look at the updated PR.  We no longer need to set `nxos_config_working_dir` as I changed to using `tempfile` instead which should be safer and avoid the possibility of collisions.